src/renderer/renderer_elements.ts

```diff
@@ -860,6 +862,152 @@ export class ReturnBlock extends BasicBlock {
+                w: 0,
+                h: 0,
+            };
+        } else {
+            clamped = {
+                x: topleft.x,
+                y: topleft.y,
+                x2: topleft.x + this.width,
+                y2: topleft.y + this.height,
+                w: 0,
+                h: 0,
+            };
+        }
+        clamped.w = clamped.x2 - clamped.x;
+        clamped.h = clamped.y2 - clamped.y;
+        if (!renderer.viewportOnly) {
+            clamped = {
+                x: topleft.x,
+                y: topleft.y,
+                x2: 0,
+                y2: 0,
+                w: this.width,
+                h: this.height,
+            };
+        }
+
+        // Draw the loop background below everything and stroke the border.
+        ctx.fillStyle = this.getCssProperty(
+            renderer, '--conditional-background-color'
+        );
+        ctx.strokeStyle = this.getCssProperty(
+            renderer, '--conditional-foreground-color'
+        );
+        ctx.fillRect(clamped.x, clamped.y, clamped.w, clamped.h);
+
+        // Only draw line if close enough.
+        const ppp = renderer.get_canvas_manager()?.points_per_pixel();
+        if (!renderer.adaptiveHiding || (ppp && ppp < SDFV.NODE_LOD))
+            ctx.strokeRect(clamped.x, clamped.y, clamped.w, clamped.h);
+
+        ctx.fillStyle = this.getCssProperty(
+            renderer, '--loop-foreground-color'
+        );
+
+        let topSpacing = LoopRegion.META_LABEL_MARGIN;
+        let remainingHeight = this.height;
+        let x = topleft.x, y = topleft.y
+        for (const [condition, region] of this.branches) {
+            topSpacing += ConditionalRegion.CONDITION_SPACING;
+            const initBottomLineY = y + ConditionalRegion.CONDITION_SPACING;
+            ctx.beginPath();
+            ctx.moveTo(x, initBottomLineY);
+            ctx.lineTo(x + this.width, initBottomLineY);
+            ctx.stroke();
+
+            if (!too_far_away_for_text(renderer)) {
+                const initTextY = (
+                    (y + (ConditionalRegion.CONDITION_SPACING / 2)) +
+                    (SDFV.LINEHEIGHT / 2)
+                );
+                const initTextMetrics = ctx.measureText(condition.string_data);
+                const initTextX = x + (initTextMetrics.width / 2);
+                ctx.fillText(condition.string_data, initTextX, initTextY);
+
+                ctx.fillText(
+                    'if', topleft.x + LoopRegion.META_LABEL_MARGIN, initTextY
+                );
+            }
+            region.draw(renderer, ctx, _mousepos)
+            y = initBottomLineY + region.height
+        }
+
+        if (visibleRect && visibleRect.x <= topleft.x &&
+            visibleRect.y <= topleft.y + SDFV.LINEHEIGHT &&
+            SDFVSettings.get<boolean>('showStateNames')) {
+            if (!too_far_away_for_text(renderer)) {
+                ctx.fillText(
+                    this.label(), topleft.x + LoopRegion.META_LABEL_MARGIN,
```

Comment:
It seems that this label is crammed in with the first condition a little bit - can you maybe add some spacing between these two?